### DISCUSSION TITLE
prisma graphql strong typing in resolver

### DISCUSSION THREAD

**Discussion by mscrocdile** [prisma/prisma#13101] [Q&A]
allusers is type [User] from graphql schema but user.findmany is prisma model type users[]
Do i understand it well that i cannot check correct type in compile time? And that there is some magic mapping?

const resolvers = {
  Query: {
    allUsers: () => {
      return prisma.user.findMany();
    }
  }
};

**Comment by nurul3101** [comment]
Hey @mscrocdile 👋 
Could you please elaborate?

`type users[]` would mean that the query returns an array of type Users.
And `[User]` in GraphQL syntax represents an array of User objects.

**Reply by mscrocdile** [reply]
Hi, 
that example above works.
I would expect that i will receive error that prisma mode result is not the same as resolver result type.
In case i will add wrong attribute into User i will notice that (maybe) in run time.
I would expect that prisma model type and graphql must be equal (e.g. lowercase).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
